### PR TITLE
feat: get and delete checkin endpoints

### PR DESCRIPTION
**Pull request by jessiiepeng**
## Implementation description. How did you make this change?
* API endpoints for the get (GET) and delete (DELETE) were created, and corresponding services `getAllCheckIns(), getCheckInsById(id: string), getCheckInsByVolunteerId(volunteerId: string), deleteCheckInById(id: string), deleteCheckInsByDateRange(startDate: string, endDate: string)`. 

## Steps to test
Get functions:
1. Insert data to PgAdmin under ```checkIn``` table (right click on checkIn table -> scripts -> INSERT Script)
2. For getting all: ```localhost:5000/checkIn``` and ensure it gets the right response
3. For getting by id: ```localhost:5000/checkIn/<id>```
4. For getting by volunteerId: ```localhost:5000/checkIn?volunteerId=<volunteerId>```
5. Make sure the get all, get by id, get by volunteer id returns the correct responses.

Delete functions:
1. insert data like above
2. For deleting by id, ```localhost:5000/checkin/<id>``` and ensures it gets deleted from table
3. For deleting by date range  ```localhost:5000//checkin?startDate=<startDate>?endDate=<endDate>``` -> make sure the parameters are correct in Postman before running the request 
^ example:  ```localhost:5000//checkin?startDate=2022-01-31T05:00:00.000Z?endDate=2022-02-31T05:00:00.000Z```

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages follow conventional commits and are descriptive. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s) 
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR
- [x] The appropriate tests if necessary have been written

### CODE REVIEW

File: backend/typescript/services/implementations/checkInService.ts
Comment (xsharonhe) [inline backend/typescript/services/implementations/checkInService.ts:132]: I think we should probably validate that the startDate is before endDate?

Comment (linnall) [reply backend/typescript/services/implementations/checkInService.ts:136]: PR #153 includes this check in the validators :)

Comment (xsharonhe) [reply backend/typescript/services/implementations/checkInService.ts:136]: ty for pointing it out :)

File: backend/typescript/middlewares/validators/checkInValidators.ts
Comment (xsharonhe) [inline backend/typescript/middlewares/validators/checkInValidators.ts:4]: planning on using linna's and dora's validators to check if start date is greater than end date

**Comment by xsharonhe** [comment]
please do not review the "do not merge" label - this ticket is meant to merge after another PR and will need to address merge conflicts after :)

File: backend/typescript/middlewares/validators/checkInValidators.ts
Comment (Shehryar21) [reply backend/typescript/middlewares/validators/checkInValidators.ts:4]: but isnt that for body contents not query params?

Comment (xsharonhe) [reply backend/typescript/middlewares/validators/checkInValidators.ts:4]: oh yeah that's true my bad - i dont think we can validate it through middlewares then (?) i think jessie has a validation in her function

File: backend/typescript/services/implementations/checkInService.ts
Comment (Shehryar21) [inline backend/typescript/services/implementations/checkInService.ts:53]: replace `scheduling` with `checkin` plz

File: backend/typescript/services/implementations/checkInService.ts
Comment (Shehryar21) [inline backend/typescript/services/implementations/checkInService.ts:95]: same here. replace scheduling with checkin plz

File: backend/typescript/rest/checkInRoutes.ts
Comment (Shehryar21) [inline backend/typescript/rest/checkInRoutes.ts:91]: should also check if they are valid date strings.

I dont think this is working right now as well because when I make the following postman call, the call just keeps loading.
image

I think it is because we are comparing strings, not date, so i think you should convert them to date first and then make those checks of whether they are valid dates and if start date is before end date.

**Review by Shehryar21 — CHANGES_REQUESTED** [review]
Overall very very nice work!! GET endpoints all worked fine for me, made some comments for delete though.

File: backend/typescript/rest/checkInRoutes.ts
Comment (jessiiepeng) [reply backend/typescript/rest/checkInRoutes.ts]: just fixed it hopefully; now we're comparing dates instead of strings by converting them first

File: backend/typescript/rest/checkInRoutes.ts
Comment (Shehryar21) [inline backend/typescript/rest/checkInRoutes.ts:89]: lets use dayjs library to deal with dates. So you can do smth like the following after importing dayjs:
```suggestion
    const startDateRange = dayjs(startDate as string);
    const endDateRange = dayjs(endDate as string);
    if (!startDateRange.isValid() || !endDateRange.isValid()) {
        // error message
    }
```
dayjs documentation reference: https://day.js.org/docs/en/parse/is-valid

Comment (jessiiepeng) [reply backend/typescript/rest/checkInRoutes.ts]: just made changes!

**Review by Shehryar21 — APPROVED** [review]
Endpoints are working as expected for me!!

**Pull request merged**
